src/test/java/seedu/address/logic/parser/FilterCommandParserTest.java

```diff
@@ -45,8 +47,34 @@ public void parse_validArgs_returnsFilterCommand() {
         criteriaPredicate = new PersonWithCriteriaPredicate(ranges);
         expected = new FilterCommand(criteriaPredicate);
         assertParseSuccess(parser, appointment, expected);
+
+        // tag
+        String tagStr = " t/ obesity";
+        String differentCaseTagStr = " t/ OBesity";
+        Tag tag = new Tag("obesity");
+        Set<Tag> tagSet = Set.of(tag);
+        criteriaPredicate = new PersonWithCriteriaPredicate(new ArrayList<>(), tagSet);
+        expected = new FilterCommand(criteriaPredicate);
+        assertParseSuccess(parser, tagStr, expected);
+        assertParseSuccess(parser, differentCaseTagStr, expected);
     }
 
+    @Test
+    public void parse_invalidRange_throwsParseException() {
+        // invalid age range
+        String age = " b/ 10-";
+        assertParseFailure(parser, age, String.format(INCORRECT_RANGE));
+        age = " b/ 10-fifty";
+        assertParseFailure(parser, age, String.format(INCORRECT_AGE));
+
+        // invalid appointment range
+        String appointment = " ap/ 01/01/2024-01-01-2026";
+        assertParseFailure(parser, appointment, String.format(INCORRECT_RANGE));
+        appointment = " ap/ 01/01/2025-12/30/2025";
+        assertParseFailure(parser, appointment, String.format(INCORRECT_DATE_FORMAT));
+
+
+    }
```

Comment:
perhaps can add a invalid tag test (e.g. Obese_) since tags are restricted to alphanumerical characters

Reply:
Would it be more suitable to include the test for isValidTagName in TagTest? I have already included some invalid tags test case in TagTest.